kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -271,25 +271,67 @@ function showExamReportDetailPage(store, classId, examId) {
   store.dispatch('CORE_SET_PAGE_LOADING', false);
 }
 
-function activateExam() {
-  this.displayModal(false);
+function activateExam(store, examId) {
+  ExamResource.getModel(examId).save({ active: true }).then(
+    () => {
+      const exams = store.state.pageState.exams;
+      const examIndex = exams.findIndex(exam => exam.id === examId);
+      exams[examIndex].active = true;
+
+      store.dispatch('SET_EXAMS', exams);
```

Comment:
In the increment example, though, it does mutate the state in place `state.count++`.

Would it be more proper to move the updates (delete, rename, etc.) to the `state` down to the mutation level instead of doing it here in the action?